lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -151,15 +151,48 @@ def sniff!
     ES2_SNIFF_RE_URL  = /([^\/]*)?\/?([^:]*):([0-9]+)/
     # Sniffs and returns the results. Does not update internal URLs!
     def check_sniff
-      url, resp = perform_request(:get, '_nodes')
+      url, resp = perform_request(:get, '_nodes/http')
```

Comment:
@jordansissel I copied that from @karmi 's code in the elasticsearch ruby client.

https://github.com/elastic/elasticsearch-ruby/blob/master/elasticsearch-transport/lib/elasticsearch/transport/transport/sniffer.rb